### ISSUE TITLE
Inconsistent and misleading resolution of multi-line elsif

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#15445]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Style/MultilineIfThen

### Steps to reproduce the problem

```ruby
# All expanded = OK
if 2 + 2 == 4
  2
elsif 2 + 2 == 5
  3
else
  4
end.then { puts it }

# One collapsed = Do not use `then` for multi-line `elsif`
if 2 + 2 == 4 then 2
elsif 2 + 2 == 5
  3
else
  4
end.then { puts it }

# Two collapsed = OK
if 2 + 2 == 4 then 2
elsif 2 + 2 == 5 then 3
else
  4
end.then { puts it }

# All collapsed = OK
if 2 + 2 == 4 then 2
elsif 2 + 2 == 5 then 3
else 4
end.then { puts it }
```

### Expected behavior

`Either expand all branches (more readable), or collapse all using "then" (more compact).`

### Actual behavior

Gives advice that, if opposed, leads to no cop, and appears generally inconsistent.

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
  TargetRubyVersion: 4.0.5
Style/FrozenStringLiteralComment:
  Enabled: false
```

### RuboCop version

```shell
System Version: macOS 26.5.1 (25F80)
Model Identifier: MacBookAir10,1
ruby 4.0.5 (2026-05-20 revision 64336ffd0e) +PRISM [arm64-darwin25]
rubocop     2026-07-10 5e56132d8
```
